File: core/src/main/java/fr/sncf/osrd/infra_state/api/InfraStateView.java
package fr.sncf.osrd.infra_state.api;

import fr.sncf.osrd.infra.api.reservation.DetectionSection;
import fr.sncf.osrd.infra.api.reservation.ReservationRoute;

public interface InfraStateView {
    /** Returns the state of a reservation section */
    DetectionSectionState getState(DetectionSection section);

    /** Returns the state of a reservation route */
    ReservationRouteState getState(ReservationRoute route);
}

File: core/src/main/kotlin/fr/sncf/osrd/dyn_infra/api/InfraStateView.kt
package fr.sncf.osrd.dyn_infra.api

import fr.sncf.osrd.infra.api.reservation.DetectionSection
import fr.sncf.osrd.infra.api.reservation.ReservationRoute

interface InfraStateView {
    /** Returns the state of a reservation section  */
    fun getState(section: DetectionSection): DetectionSectionState

    /** Returns the state of a reservation route  */
    fun getState(route: ReservationRoute): ReservationRouteState
}